data/account/participationRegistry.go

```diff
@@ -332,61 +407,163 @@ func (db *participationDB) GetAll() (records []ParticipationRecord, err error) {
 	return
 }
 
+func (db *participationDB) Get(id ParticipationID) (record ParticipationRecord, err error) {
+	db.mutex.RLock()
+	defer db.mutex.RUnlock()
+
+	record, ok := db.cache[id]
+	if !ok {
+		return ParticipationRecord{}, ErrParticipationIDNotFound
+	}
+	return record, nil
+}
+
+func (db *participationDB) GetAll() ([]ParticipationRecord, error) {
+	db.mutex.RLock()
+	defer db.mutex.RUnlock()
+
+	results := make([]ParticipationRecord, 0, len(db.cache))
+	for _, record := range db.cache {
+		copy := record
+		results = append(results, copy)
+	}
+	return results, nil
+}
+
+// updateRollingFields sets all of the rolling fields according to the record object.
+func (db *participationDB) updateRollingFields(ctx context.Context, tx *sql.Tx, record ParticipationRecord) error {
+	_, err := tx.ExecContext(ctx, updateRollingFields,
```

Comment:
check the number of affected records. You know that it should always result in a single row update, so you should always test for that.